File: EnlightenmentApp.ModuleService/EnlightenmentApp.BLL/Items/Module.cs
namespace EnlightenmentApp.BLL.Entities
{
    public class Module : BaseItem
    {
        #nullable disable
        public string Title { get; set; }
        public string Author { get; set; }
        public float Rating { get; set; }
        public int Cost { get; set; }
        public string Summary { get; set; }
        public ICollection<ModuleReview> Reviews { get; set; }
        public ICollection<Tag> Tags { get; set; }
        public string ImageSrc { get; set; }
        public ICollection<Path> Paths { get; set; }
        public ICollection<Section> Sections { get; set; }
        public bool IsCompleted { get; set; }
    }
}

File: EnlightenmentApp.FrontendApp/frontend-react/src/data/models/Module/Module.ts
import Tag from "../Tag/Tag";
import Section from "../Section/Section";
import ModuleReview from "../ModuleReview/ModuleReview";

export default class Module {
  id: number = 0;
  title: string = "";
  author: string = "";
  rating: number = 0;
  cost: number = 0;
  summary: string = "";
  reviews: ModuleReview[] = [];
  tags: Tag[] = [];
  imageSrc: string = "";
  sections: Section[] = [];
  isCompleted: boolean = false;
}
